cnf-tests/testsuites/e2esuite/multinetworkpolicy/multinetworkpolicy_sriov.go

```diff
@@ -0,0 +1,750 @@
+
+	podA := pods.DefinePod(namespace)
+	pods.RedefineWithLabel(podA, "pod", "a")
+	pods.RedefinePodWithNetwork(podA, TEST_NETWORK)
+	redefineWithNetcatServers(podA)
+	_, err := pods.RedefineAsPrivileged(podA, "")
+	Expect(err).To(BeNil())
+	podA.ObjectMeta.GenerateName = "testpod-a-"
+	podA = createAndStartPod(podA)
+
+	podB := pods.DefinePod(namespace)
+	pods.RedefineWithLabel(podB, "pod", "b")
+	pods.RedefinePodWithNetwork(podB, TEST_NETWORK)
+	redefineWithNetcatServers(podB)
+	_, err = pods.RedefineAsPrivileged(podB, "")
+	Expect(err).To(BeNil())
+	podB.ObjectMeta.GenerateName = "testpod-b-"
+	podB = createAndStartPod(podB)
+
+	podC := pods.DefinePod(namespace)
+	pods.RedefineWithLabel(podC, "pod", "c")
+	pods.RedefinePodWithNetwork(podC, TEST_NETWORK)
+	redefineWithNetcatServers(podC)
+	_, err = pods.RedefineAsPrivileged(podC, "")
+	Expect(err).To(BeNil())
+	podC.ObjectMeta.GenerateName = "testpod-c-"
+	podC = createAndStartPod(podC)
+
+	return podA, podB, podC
+}
+
+func waitForObject(clientSet *sriovtestclient.ClientSet, key runtimeclient.ObjectKey, object runtimeclient.Object) {
+
+	Eventually(func() error {
+		return clientSet.Get(context.Background(), key, object)
+	}, 120*time.Second, 1*time.Second).
+		WithOffset(1).
+		ShouldNot(
+			HaveOccurred(),
+			"Object [%s] not found for key [%s]", reflect.TypeOf(object), key,
+		)
+}
+
+// Add 4 containers to the pod for netcat server. 2 for each transport protocol (TCP/UDP) for each
+// port in (5555/6666)
+func redefineWithNetcatServers(pod *corev1.Pod) *corev1.Pod {
+	// --keep-open parameters helps to use the same server for multiple tests. Incoming messages
+	// are forwarder to pod logs (stdout/stderr) and they are used to validate connectivity (see canSendTraffic)
+	// UDP servers support --keep-open only with --sh-exec option, and to get the incoming messages
+	// to logs they are needed to be sent to stderr, as stdout is redirected back to the client
+	pod.Spec.Containers = append(pod.Spec.Containers, corev1.Container{
+		Name:    "netcat-tcp-server-5555",
+		Image:   images.For(images.TestUtils),
+		Command: []string{"nc", "--keep-open", "--listen", "0.0.0.0", "5555"}})
+	pod.Spec.Containers = append(pod.Spec.Containers, corev1.Container{
+		Name:    "netcat-udp-server-5555",
+		Image:   images.For(images.TestUtils),
+		Command: []string{"nc", "--keep-open", "--udp", "--sh-exec", "/bin/cat >&2", "--listen", "0.0.0.0", "5555"}})
+	pod.Spec.Containers = append(pod.Spec.Containers, corev1.Container{
+		Name:    "netcat-tcp-server-6666",
+		Image:   images.For(images.TestUtils),
+		Command: []string{"nc", "--keep-open", "--listen", "0.0.0.0", "6666"}})
+	pod.Spec.Containers = append(pod.Spec.Containers, corev1.Container{
+		Name:    "netcat-udp-server-6666",
+		Image:   images.For(images.TestUtils),
+		Command: []string{"nc", "--keep-open", "--udp", "--sh-exec", "/bin/cat >&2", "--listen", "0.0.0.0", "6666"}})
+	return pod
+}
+
+func createAndStartPod(pod *corev1.Pod) *corev1.Pod {
```

Comment:
I think we have a function for this in the pkg folder please have a look

Reply:
Can't find anything similar. Can you provide any other details?